include/pybind11/cast.h

```diff
@@ -1036,10 +1057,8 @@ struct type_caster<T, enable_if_t<std::is_arithmetic<T>::value && !is_std_char_t
 
         bool py_err = py_value == (py_type) -1 && PyErr_Occurred();
 
-        // Protect std::numeric_limits::min/max with parentheses
```

Comment:
It's poorly documented, but yes, I think more recent NVCC versions support suppression if you can find the right code. But this will actually generate simpler/faster _runtime code_ with 2 fewer checks, if this happens fairly often.

I also think the logic may be flawed; it might be good to rework with new logic. Also I'm surprised it's showing up with some very odd types; things like pointers were `static_cast` to an integer is not allowed (which is why it's using C-style casts); I guess this is used to covert pointer addresses to Python ints?

Reply:
Check the new version out! Simpler logic (the tests here are pretty comprehensive, don't ask me how I know that 😊), with no more pointless comparisons.